### DISCUSSION TITLE
[Question]: I used openid for azure ad login but error.How to solve?

### DISCUSSION THREAD

**Discussion by angpao** [danny-avila/LibreChat#4219] [Troubleshooting]
### What is your question?

{"level":"error","message":"[openidStrategy] Key 'roles' not found in id token!"}

### More Details

{"level":"error","message":"[openidStrategy] Key 'roles' not found in id token!"}

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by angpao** [comment]
I follow up as https://www.librechat.ai/docs/configuration/authentication/OAuth2-OIDC/azure to restrict access by groups.

**Comment by terence4444** [comment]
I spent several hours and found the configuration guide has minor errors.
With regarding the .env config, you have to remove all the quotes.
OPENID_REQUIRED_ROLE=(Here must be App roles-> role VALUE, not Display name)  
OPENID_REQUIRED_ROLE_TOKEN_KIND=id  
OPENID_REQUIRED_ROLE_PARAMETER_PATH=roles
